examples/ssh/opkssh/cli.go

```diff
@@ -0,0 +1,308 @@
+package main
+
+import (
+	"crypto"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/fs"
+
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/exp/slices"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/openpubkey/openpubkey/examples/ssh/sshcert"
+	"github.com/openpubkey/openpubkey/parties"
+	"github.com/openpubkey/openpubkey/pktoken"
+	"github.com/openpubkey/openpubkey/util"
+)
+
+// This function is called by the SSH server as the AuthorizedKeysCommand:
+//
+// The following lines are added to /etc/ssh/sshd_config:
+//
+//	AuthorizedKeysCommand /etc/opk/opkssh ver %u %t %k
+//	AuthorizedPrincipalsCommandUser root
+//
+// The parameters specified in the config map the parameters send the function below. They are:
+// %u The username (requested principal) - userArg
+// %t The public key type - typArg - in this case a certificate being used as a public key
+// %k The base64-encoded public key for authentication - certB64Arg - the public key is also a certificate
+func AuthorizedKeysCommand(userArg string, typArg string, certB64Arg string, policyEnforcer PolicyCheck, op parties.OpenIdProvider) (string, error) {
```

Comment:
The sshd_config man page calls them TOKENS (all caps).

I want to distinguish them as coming from sshd, whereas policyEnforcer and op are external.

Reply:
I've added a comment to clarify this point

Reply:
why not clarify using the variable names? userArg -> user, typArg -> publicKeyType, cert64 -> sshCert. You can still have a comment that explains where these things come from but you should understand what they are from the name itself.

Reply:
because it isn't just the user, but the user as specified by the openssh, it isn't just a sshCert but the sshCert in the B64 encoded format as understood by the openssh component of openbsd. It is important that these parameters are labeled as different and special because they have to be handled in a special manner. I had to spent four hours reading c code to get this work. It looks nice now but it was finicky. This is why I have a test with the exact output produced by openssh's c code.